### PR TITLE
[Core] remove old Constraint-B&S

### PR DESCRIPTION
**Pull request by philbucher**
this PR removes the first version of the Constraint B&S (which is broken)
The handling of the Constraints is already for a while integrated into the normal block-B&S
Furthermore it is no longer needed for the ChimeraApp

@loumalouomega I saw that you use it in the ContactApp, can you please have a look?

closes #4674

### CODE REVIEW

**Comment by loumalouomega** [comment]
In my case I just derivate from it, it just requires to remove the templating from it.

**Comment by loumalouomega** [comment]
Should be done here?

**Comment by philbucher** [comment]
> Should be done here?

sure, why not

**Comment by loumalouomega** [comment]
I can't see where is used, I think I already remove it

**Comment by philbucher** [comment]
not sure but probably you could remove the [template argument](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/ContactStructuralMechanicsApplication/custom_strategies/custom_builder_and_solvers/contact_residualbased_block_builder_and_solver.h#L60) now



> I can't see where is used, I think I already remove it

take a look at the changes, some things I removed already in Contact

File: kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp [hunk 1]
```diff
@@ -571,141 +536,6 @@ namespace Kratos
             KRATOS_CHECK_LESS_EQUAL(std::abs((rA(21,21) - 185056985.8178827166557312)/rA(21,21)), tolerance);
         }
 
-        /**
-         * Checks if the block builder and solver with constraints performs correctly the assemble of the system
-         */
-        KRATOS_TEST_CASE_IN_SUITE(ExtendedDisplacementBlockBuilderAndSolverWithConstraints, KratosCoreFastSuite)
```
Comment (loumalouomega) [inline kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp:577]: I would comment this until is fixed, the test takes some time to create...

**Review by loumalouomega — DISMISSED** [review]
Ok,I could not find it because you already removed it. It is Ok I haven't used since stopped to be default

File: kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp (hunk 1 above)
Comment (philbucher) [reply kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp:577]: I though this is a duplicate

It used the B&S that is removed in this PR

I can try to use the baseclass

**Review by adityaghantasala — DISMISSED** [review]
This is overdue .. ! 
@philbucher  thanks for deleting it !

File: kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp (hunk 1 above)
Comment (philbucher) [reply kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp:577]: I brought back the tests and made a comment

I checked, the results are not the same when I use the baseclass (which was expected), but I don't feel confident updating them

**Comment by philbucher** [comment]
can someone reapprove?

**Review by adityaghantasala — APPROVED** [review]

**Pull request merged**